compiler/src/dmd/statementsem.d

```diff
@@ -1960,6 +1960,20 @@ package (dmd) extern (C++) final class StatementSemanticVisitor : Visitor
         // Save 'root' of two branches (then and else) at the point where it forks
         CtorFlow ctorflow_root = scd.ctorflow.clone();
 
+        /* Rewrite `if (!__ctfe) A else B` as `if (__ctfe) B else A`
+         */
+        NotExp notExp;
+        if (ifs.elsebody &&
+            (notExp = ifs.condition.isNotExp()) !is null &&
+            notExp.e1.isVarExp() &&
+            notExp.e1.isVarExp().var.ident == Id.ctfe)
```

Comment:
You could apply the same strategy for the VarExp as for the NotExp:

```d
NotExp notExp;
VarExp varExp;
if (ifs.elsebody &&
     (notExp = ifs.condition.isNotExp()) !is null &&
     (varExp = notExp.e1.isVarExp() !is null &&
      varExp.e1.isVarExp().var.ident == Id.ctfe)
```

Reply:
Yes. What I'd really like to write is:
```
if (ifs.elsebody && ifs.condition.isNotExp().e1.isVarExp().var.ident == Id.ctfe)
```

Reply:
The `static if (__ctfe)` would still require both arms to be parsed and semantically analyzed by the compiler, which would be just as inelegant.

We're using the `if (__ctfe)` special case now because we didn't anticipate this problem. In hindsight, it should have been restricted as you suggest. But I've grepped for usage of `__ctfe` and found that restricting it would break a non-trivial amount of code.

I am considering extending things to support `__ctfe ? A : B` as well, it wouldn't be hard. You can add it if you want some fun!